### PR TITLE
feat: upgrade to TX EDC 0.7

### LINKED ISSUE #26: Add `helm upgrade` workflow
The current [Lint and Test Charts](https://github.com/eclipse-tractusx/knowledge-agents-edc/blob/main/.github/workflows/helm-chart-lint.yml) workflow should be enhanced by a `helm upgrade` step, that verifies, that an upgrade from the latest released Chart version to the current chart present in the repo is possible.

This is not blocking the current release for Tractus-X 3.2, since there was not knowledge-agents-edc release before anyways, but should be added for future checks.

### LINKED ISSUE #30: [TRG 5.04] Define resource limit/request for Charts
In our release guidelines, that are defined for the whole Eclipse Tractus-X project, we specify, that each Chart should provide
default resource -request and -limit values.
In best case the values are gathered by monitoring the EDC under different load circumstances. This would give users a good initial understanding or resource consumption.

There are already request and limit values specified [as comments in the values.yaml](https://github.com/eclipse-tractusx/knowledge-agents-edc/blob/main/charts/agent-connector-azure-vault/values.yaml#L255). 
To follow the release guideline, they could simply be used as actual defaults.

If there are valid reasons, why resource requests and limits cannot be set, we should change the [existing TRG 5.04](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-04/), so the guidelines are followed and we do not need to track exceptions.

### LINKED ISSUE #31: Clarification: Deployability of Helm Chart with default config
## Description

The description of [TRG 5.01](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-01/) contains a section, that Charts should be installable on any environment with just the config provided by the default `values.yaml`.

I tried that with all of the Charts contained in this repo and while I can see, that `tractusx-dev/agent-connector-azure-vault --version 1.9.6-SNAPSHOT` would most likely need config in any circumstance, I was expecting the `agent-connector-memory` to work without further config.

blocks #17 

## Additional info

All of the tests where done with the released Charts in version `1.9.6-SNAPSHOT`

### LINKED ISSUE #61: Dedicated helm chart for the Dataplane only?
As the discussion area is not available in this repo, I apologise for starting a discussion here.

I am currently looking at the knowledge agent as we as ZF want to provide this in relation to a Catena-X/Tractus-X use case. As far as I understand, the knowledge agent magic is contained in an EDC extension which is to be placed in the dataplane. As I like decoupled systems, I would now expect there to be a helm chart that just provides an additional dataplane that can simply be installed alongside an already installed EDC in K8s. Looking through the Helm Charts provided, I only found copies of the tractusx-edc helm charts with some additional values added for the knowledge-agents. Why was this approach taken? This approach does result in the knowlege-agent team having to painstakingly adopt the changes of the tractusx-edc helm chart for each new tractusx-edc release, doesn't it?

Regarding my expectation I would think of something like the following:

```mermaid
flowchart TB
    subgraph tractusx-edc Helm Chart
        Controlplane-->Dataplane
    end

    subgraph KA Helm Chart
    KA-Dataplane --> Controlplane
    end
```

And of course, for a simple all-in-one setup, you could also consider having the tractusx-edc helm chart as a dependency in the knowledge-agent helm chart:

```mermaid
flowchart TB
    subgraph KA Helm Chart
        subgraph tractusx-edc Helm Chart
            Controlplane-->Dataplane
        end
    KA-Dataplane --> Controlplane
    end
```

What do you think about these points? Maybe in this issue we could collect all the necessary steps that would allow such a simple and easy setup, and/or of course other possible ways to avoid copying and pasting code from other projects that needs to be done manually.

### LINKED ISSUE #94: Story "Upgrade to TX EDC 0.6.0"
# Story "Upgrade to TX EDC 0.6.0"

Repository: https://github.com/eclipse-tractusx/knowledge-agents-edc

## Statement 

As a (potential) KA participant, I want the KA Agent Plane to be compiled and deployed against EDC 0.6.0 and being aligned with its latest features such that I can seamlessly replace my existing EDC deployment/KA-enabled EDC deployment with the new version

## Acceptance Criteria

- new version of the helm charts for each variant that EDC provides
- dependencies adapted to TX EDC 0.6.0 / EDC ?0.3.1?
- integration test runs

## Estimation 

3 SP

_Originally posted by @drcgjung in https://github.com/eclipse-tractusx/sig-release/issues/287#issuecomment-1770539562_

### LINKED ISSUE #97: Story "Replace/Invoke Matchmaking Agent API"
# Story "Replace/Invoke Matchmaking Agent API"

Repository https://github.com/eclipse-tractusx/knowledge-agents-edc

## Statement 

As KA user, I want to replace existing code with an invocation of the Matchmaking API in order to regain a working agent plane.

## Acceptance Criteria

- references to carved out code from (agent-plane/src/main/.../sparql and agent-plane/src/main/.../rdf) to use the stubs.
- unit tests for the most common use cases

## Estimation 

10 SP

_Originally posted by @drcgjung in https://github.com/eclipse-tractusx/sig-release/issues/276#issuecomment-1770758420_

### LINKED ISSUE #98: Story "Implement Data-Plane Chart"
# Story "Implement Data-Plane Chart"

Repository: https://github.com/eclipse-tractusx/knowledge-agents-edc

## Statement 

As a (potential) KA participant, I want the KA Agent Plane to be deployable alongside EDC charts such that I can seamlessly extend my existing EDC deployment.

## Acceptance Criteria

- find out how the dataplane can register/deregister itself in the control plane
- implement startup/teardown actions in the agent plane deployment
- make the agent plane chart refer to values in the umbrella chart
- extend the ct test 
- remove the old charts

## Estimation 

10 SP

_Originally posted by @drcgjung in https://github.com/eclipse-tractusx/sig-release/issues/286#issuecomment-1770924931_

### LINKED ISSUE #129: TRG 5.09 Helm test and upgrade workflow missing
The current [helm-chart-lint](https://github.com/eclipse-tractusx/knowledge-agents-edc/blob/main/.github/workflows/helm-chart-lint.yml) workflow is only linting the Chart, but not executing the test and upgrade step, like described in the workflow example in [TRG 5.09](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-09)

### LINKED ISSUE #179: TRG 5.11 Upgradeability

### LINKED ISSUE #185: broken links in README
https://github.com/eclipse-tractusx/knowledge-agents-edc
has a broken link:
https://github.com/eclipse-tractusx/knowledge-agents-edc/blob/main/gent-plane/agent-plane-protocol/src/main/java/org/eclipse/tractusx/agents/edc/rdf/RDFStore.java

The correct link is
https://github.com/eclipse-tractusx/knowledge-agents-edc/blob/main/agent-plane/agent-plane-protocol/src/main/java/org/eclipse/tractusx/agents/edc/rdf/RdfStore.java
(There are two typos).

Please check all other links as well, thanks!

### PR DESCRIPTION
**Pull request by drcgjung**
## WHAT

Upgrades Agent Plane(s) to TX EDC 0.7.0 (and Apache Jena 4.9)
Switches from full connector charts to data/agent plane only chart.

## WHY

TX EDC 0.7.0 is the breaking-change release as of CX 24.05
Apache Jena has provided a new release which also updates quite some dependencies.
Having agent-plane only charts makes it easier for sidecar deployments with EDC.
Furthermore we can thus successfully test the data plane chart alone which should be now compliant to the TRGHs.
This PR has been aligned to the provider-part of the 24.05 E2E test 

## FURTHER NOTES

Closes #94 
Closes #179 
Closes #129 
Closes #98 
Closes #97 
Closes #61 
Closes #31 
Closes #30 
Closes #26 
Closes #185

### CODE REVIEW

**Review by Martin0815bla — APPROVED** [review]

**Review by obalandi — APPROVED** [review]
LGTM

**Review by obalandi — APPROVED** [review]
LGTM

File: charts/agent-plane-azure-vault/ci/integration-values.yaml [hunk 1]
```diff
@@ -22,21 +22,11 @@
...
-  postgresql: true
 
 controlplane:
   endpoints:
     management:
       authKey: "bla"
-  ssi:
-    miw:
-      url: "https://managed-identity-wallets.int"
-      authorityId: "BPNL0000000DUMMY"
```
[the remark this answers is not in the document]
Comment (drcgjung) [reply charts/agent-plane-azure-vault/ci/integration-values.yaml]: no, they are dummies - obsolete anyway by the new iatp settings.

File: charts/agent-plane/ci/integration-values.yaml [hunk 2]
```diff
@@ -22,21 +22,11 @@
...
-  postgresql: true
 
 controlplane:
   endpoints:
     management:
       authKey: "bla"
-  ssi:
-    miw:
-      url: "https://managed-identity-wallets.int"
-      authorityId: "BPNL0000000DUMMY"
```
[the remark this answers is not in the document]
Comment (drcgjung) [reply charts/agent-plane/ci/integration-values.yaml:34]: no, see iatp settings in the lastest commits.

**Comment by drcgjung** [comment]
@almadigabor The chart linting actually should go through, see https://github.com/catenax-ng/tx-knowledge-agents-edc/actions/runs/9035264222 (where we have already integrated this PR) - sometimes helm does not "know" the standard repos as it seems when doing "cross-repo" actions.

However, the dependencies check will show the need to issue an IP check for the new versions of the depending modules, as Apache Jena https://github.com/catenax-ng/tx-knowledge-agents-edc/actions/runs/9035264223

It would be great if you could issues that check.

**Comment by almadigabor** [comment]
Hi @drcgjung! I've ran the dash tool and got the following output:

```
[INFO] This content is either not correctly mapped by the system, or requires review.
[INFO] A review is required for maven/mavencentral/org.apache.jena/jena-base/4.9.0.
[INFO] A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14713 .
[INFO] A review is required for maven/mavencentral/org.apache.jena/jena-rdfpatch/4.9.0.
[INFO] A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14714 .
[INFO] A review is required for maven/mavencentral/org.apache.jena/jena-core/4.9.0.
[INFO] A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14700 .
[INFO] A review is required for maven/mavencentral/org.apache.jena/jena-tdb2/4.9.0.
[INFO] A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14701 .
[INFO] A review is required for maven/mavencentral/org.apache.jena/jena-fuseki-core/4.9.0.
[INFO] A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14702 .
[INFO] A review is required for maven/mavencentral/org.apache.jena/jena-rdfconnection/4.9.0.
[INFO] A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14703 .
[INFO] A review is required for maven/mavencentral/org.apache.jena/jena-dboe-storage/4.9.0.
[INFO] A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14704 .
[INFO] A review is required for maven/mavencentral/org.apache.jena/jena-dboe-base/4.9.0.
[INFO] A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14705 .
[INFO] A review is required for maven/mavencentral/org.apache.jena/jena-shacl/4.9.0.
[INFO] A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14706 .
[INFO] A review is required for maven/mavencentral/org.apache.jena/jena-dboe-trans-data/4.9.0.
[INFO] A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14707 .
[INFO] A review is required for maven/mavencentral/org.apache.jena/jena-tdb/4.9.0.
[INFO] A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14708 .
[INFO] A review is required for maven/mavencentral/org.apache.jena/jena-shex/4.9.0.
[INFO] A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14709 .
[INFO] A review is required for maven/mavencentral/org.apache.jena/jena-dboe-index/4.9.0.
[INFO] A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14710 .
[INFO] A review is required for maven/mavencentral/org.apache.jena/jena-arq/4.9.0.
[INFO] A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14711 .
[INFO] A review is required for maven/mavencentral/org.apache.jena/jena-dboe-transaction/4.9.0.
[INFO] A review request already exists https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14712 .
[INFO] Summary file was written to: /Users/A200134305/repos/products/tx-knowledge-agents-edc/DEPENDENCIES
```

Let's wait until the afternoon for these to be resolved then I can review and merge your PR. You might need a DEPENDENCIES file update once the IP issues are closed.

**Comment by almadigabor** [comment]
Hey @drcgjung, can you update the dependencies file? If I see it correctly, all issues have been approved.

**Comment by drcgjung** [comment]
> Hey @drcgjung, can you update the dependencies file? If I see it correctly, all issues have been approved.

@almadigabor done and successful. thx.

**Comment by almadigabor** [comment]
I opened an IP review issue due to the large number of changes. Please track the status [here](https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/14721).

File: charts/agent-plane-azure-vault/templates/_helpers.tpl [hunk 3]
```diff
@@ -1,9 +1,9 @@
 #
-#  Copyright (c) 2023 T-Systems International GmbH
+#  Copyright (c) 2023,2024 T-Systems International GmbH
 #  Copyright (c) 2023 ZF Friedrichshafen AG
 #  Copyright (c) 2023 Mercedes-Benz Tech Innovation GmbH
 #  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
-#  Copyright (c) 2021, 2023 Contributors to the Eclipse Foundation
+#  Copyright (c) 2021, 2024 Contributors to the Eclipse Foundation
```
Comment (almadigabor) [inline charts/agent-plane-azure-vault/templates/_helpers.tpl:6]: Please make sure you don't have a white space between the years in the license header like [here](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02#copyright-and-license-header).

File: charts/agent-plane/templates/_helpers.tpl [hunk 4]
```diff
@@ -1,9 +1,9 @@
 #
-#  Copyright (c) 2023 T-Systems International GmbH
+#  Copyright (c) 2023,2024 T-Systems International GmbH
 #  Copyright (c) 2023 ZF Friedrichshafen AG
 #  Copyright (c) 2023 Mercedes-Benz Tech Innovation GmbH
 #  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
-#  Copyright (c) 2021, 2023 Contributors to the Eclipse Foundation
+#  Copyright (c) 2021, 2024 Contributors to the Eclipse Foundation
```
Comment (almadigabor) [inline charts/agent-plane/templates/_helpers.tpl:6]: Same here, please check all the files in the PR to have a proper license header format.

**Review by almadigabor — CHANGES_REQUESTED** [review]
Hey @drcgjung, see my comments below.

File: charts/agent-plane/templates/_helpers.tpl (hunk 4 above)
Comment (drcgjung) [reply charts/agent-plane/templates/_helpers.tpl]: went over the complete directory using a search expression. Hope I got all

Comment (drcgjung) [reply charts/agent-plane/templates/_helpers.tpl]: had to mitigate some SAST issues at the same time (and did the same also for the https://github.com/eclipse-tractusx/knowledge-agents/pull/120

**Comment by drcgjung** [comment]
Again, the chart testing process somehow stuck but should be successful, if its been restarted: see https://github.com/catenax-ng/tx-knowledge-agents-edc/actions/runs/9138105951

**Comment by almadigabor** [comment]
The IP check issue have been approved.

**Review by almadigabor — APPROVED** [review]

**Pull request merged**